common/flatpak-context.c

```diff
@@ -888,6 +888,114 @@ flatpak_context_add_nousb_query (FlatpakContext  *context,
   flatpak_context_add_query_to (context->blocked_usb_devices, usb_query);
 }
 
+static gboolean
+flatpak_context_add_usb_list (FlatpakContext *context,
+                              const char     *list,
+                              GError        **error)
+{
+  if (!flatpak_context_parse_usb_list (list, context->allowed_usb_devices,
+                                       context->blocked_usb_devices, error))
+    return FALSE;
+
+  return TRUE;
+}
+
+static gboolean
+flatpak_context_add_usb_list_from_file (FlatpakContext *context,
+                                        const char     *path,
+                                        GError        **error)
+{
+  g_autofree char *contents = NULL;
+
+  if (!flatpak_validate_path_characters (path, error))
+    return FALSE;
+  if (!g_file_get_contents (path, &contents, NULL, error))
+    return FALSE;
+
+  if (!flatpak_context_parse_usb_list (contents, context->allowed_usb_devices,
+                                       context->blocked_usb_devices, error))
+    return FALSE;
+
+  return TRUE;
+}
+
+gboolean
+flatpak_context_parse_usb_list (const char     *buffer,
+                                GHashTable     *allowed,
+                                GHashTable     *blocked,
+                                GError        **error)
+{
+  char *line = NULL;
+  g_autoptr(FlatpakUsbQuery) usb_query = NULL;
+  g_autoptr(GInputStream) stream = NULL;
+  g_autoptr(GDataInputStream) buffered = NULL;
+
+  stream = g_memory_input_stream_new_from_data (buffer, -1, NULL);
+  if (!stream)
+    return FALSE;
+
+  buffered = g_data_input_stream_new (G_INPUT_STREAM (stream));
+  while ((line = g_data_input_stream_read_line (buffered, NULL, NULL, error)))
+    {
+      g_auto(GStrv) split = NULL;
+      GStrv current = NULL;
+      gboolean blocking = FALSE;
+
+      if (line[0] == '#')
+        {
+          g_free (line);
+          continue;
+        }
```

Comment:
Small idea for a cleanup: instead of manually calling `g_free` in various branches, declare an autofree variable inside the while loop:

```c
while ((char *aux = g_data_input_stream_read_line (buffered, NULL, NULL, error)))
  {
    g_autofree char *line = g_steal_pointer (&aux);
    [...]
```

This should make this a little more robust